keps/sig-auth/2862-fine-grained-kubelet-authz/README.md

```diff
@@ -0,0 +1,1013 @@
+exploited by an attacker to escalate privilege.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Introduce new subresources for `/healthz` and `/pods/` kubelet endpoints,
+allowing for more  granular authorization.
+- These changes should be backwards compatible.
+- These changes should not break users on upgrade.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Create a new Kubelet API.
+- Remove the existing subresources.
+- Make the kubelet API node-restricted.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Add new subresources for `healthz` and `pods` endpoint while supporting the old coarse grained
+proxy subresources as shown in the table below.
+
+| Request path                                     | Existing Subresource | Proposed Subresource(s) |
+| ----------------------------------------------- | -------------------- | ------------------------ |
+| /healthz                                        | proxy                | proxy, healthz           |
+| /healthz/log                                    | proxy                | proxy, healthz           |
+| /healthz/ping                                   | proxy                | proxy, healthz           |
+| /healthz/syncloop                               | proxy                | proxy, healthz           |
+| /pods/                                          | proxy                | proxy, pods              |
+| /runningpods/                                   | proxy                | proxy, pods              |
+
+This way users who were previously using the `proxy` subresource to grant access to the
+`/healthz` kubelet endpoint can update their ClusterRole or Role by replacing the
+`proxy` subresource with the `healthz` subresource.
+
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a security conscious node monitoring agent owner I want to interact with the
+kubelet API to list pods on the node without having to grant the agent the `proxy` subresource and use
+least privilege to grant the exact permissions the agent needs like `pods` or 
+`healthz`.
+
+### Notes
+
+If this change were to be implemented as proposed, the `proxy` `subresource` 
+would still cover `/attach/`, `/exec/`, `/run/`, `/configz/`, `/debug/`. 
```

Comment:
what is the motivation to keep `/configz/`under `proxy`? Reading config maybe useful for monitoring tools

Reply:
+1 for separating configz

Reply:
Done. Thanks!